common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -536,9 +619,11 @@ public MergeStatuses finalizeShuffleMerge(FinalizeShuffleMerge msg) {
         }
       }
       // Even when the mergePartitionsInfo is null, we mark the shuffle as finalized but the results
-      // sent to the driver will be empty. This cam happen when the service didn't receive any
+      // sent to the driver will be empty. This can happen when the service didn't receive any
       // blocks for the shuffle yet and the driver didn't wait for enough time to finalize the
       // shuffle.
+      writeAppAttemptShuffleMergeInfo(
```

Comment:
We are writing this to db even before we actually finalize the partition, that is, close all the partition files? What happens when there are error during finalizing the files?

Reply:
Moved the write prior to the final return in this method.